Sources/BraintreeDropIn/BTPaymentSelectionViewController.m

```diff
@@ -364,89 +217,47 @@ - (void)sendVaultedCardAppearAnalytic {
-                                                         error:nil];
-
-        if ([cell.paymentMethodNonce isKindOfClass:[BTCardNonce class]]) {
-            [self.apiClient sendAnalyticsEvent:@"ios.dropin2.vaulted-card.select"];
-        }
-    }
-}
-
 #pragma mark UITableViewDelegate
 
 - (NSInteger)numberOfSectionsInTableView:(__unused UITableView *)tableView {
-    return 1;
+    return self.hasVaultedPaymentMethods ? 2 : 1;
 }
 
 - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
-    static NSString *simpleTableIdentifier = @"BTDropInPaymentSeletionCell";
+    if (indexPath.section == 0 && self.hasVaultedPaymentMethods) {
+        static NSString *identifier = @"BTVaultedPaymentMethodsTableViewCell";
 
-    BTDropInPaymentSeletionCell *cell = [tableView dequeueReusableCellWithIdentifier:simpleTableIdentifier forIndexPath:indexPath];
+        BTVaultedPaymentMethodsTableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:identifier forIndexPath:indexPath];
+        cell.paymentMethodNonces = self.paymentMethodNonces;
+        cell.delegate = self;
+        return cell;
+    } else {
+        static NSString *simpleTableIdentifier = @"BTDropInPaymentSelectionCell";
 
-    cell.accessoryType = UITableViewCellAccessoryNone;
-    cell.selectionStyle = UITableViewCellSelectionStyleDefault;
-    BTUIKPaymentOptionType option = ((NSNumber*)self.paymentOptionsData[indexPath.row]).intValue;
+        BTDropInPaymentSelectionCell *cell = [tableView dequeueReusableCellWithIdentifier:simpleTableIdentifier forIndexPath:indexPath];
 
-    cell.detailLabel.text = @"";
-    cell.label.text = [BTUIKViewUtil nameForPaymentMethodType:option];
-    if (option == BTUIKPaymentOptionTypeUnknown) {
-        cell.label.text = BTUIKLocalizedString(CREDIT_OR_DEBIT_CARD_LABEL);
-    }
-    cell.iconView.paymentOptionType = option;
-    cell.type = option;
+        cell.accessoryType = UITableViewCellAccessoryNone;
+        cell.selectionStyle = UITableViewCellSelectionStyleDefault;
+        BTUIKPaymentOptionType option = ((NSNumber*)self.paymentOptionsData[indexPath.row]).intValue;
+
+        cell.detailLabel.text = @"";
+        cell.label.text = [BTUIKViewUtil nameForPaymentMethodType:option];
+        if (option == BTUIKPaymentOptionTypeUnknown) {
+            cell.label.text = BTUIKLocalizedString(CREDIT_OR_DEBIT_CARD_LABEL);
+        }
+        cell.iconView.paymentOptionType = option;
+        cell.type = option;
 
-    return cell;
+        return cell;
+    }
 }
 
 - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
-    BTDropInPaymentSeletionCell *cell = [tableView cellForRowAtIndexPath:indexPath];
+    if (![[tableView cellForRowAtIndexPath:indexPath] isKindOfClass:BTDropInPaymentSelectionCell.class]) {
+        return;
+    }
+
+    BTDropInPaymentSelectionCell *cell = [tableView cellForRowAtIndexPath:indexPath];
```

Comment:
I think we also need to handle the case where a `BTVaultedPaymentMethodsTableViewCell` is selected in this method

Reply:
AH NVM - I see now that the vaulted cell contains a UICollectionView